### DISCUSSION TITLE
Help configure iptables firewall for minimizing access to VM only for self hosted Budibase

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#10725] [Help]
Hi,

We have deployed a self hosted Budibase at a dedicated VM and budibase is set to listen to default port 10000. We would like to isolate access to it as much as possible. I am trying to configure iptables to allow only incoming access at port 10000 but when I activate ip tables I can only an error screen of Budibase.  Budibase is installed with docker compose installation.

What ports need to be open in order for Budibase to work properly?
Do you have a sample configuration to suggest?

Thank you!

**Comment by jonnymccullagh** [comment]
Hi @gmentzik, we don't currently have any documentation for iptables or firewalld but that is something we could consider adding in the coming months. We could prioritise this if you are on business support.
That said, I think you should be able to get iptables working with port 10000 opened so long as nginx on port 10000 is able to access the services on the other ports (e.g. 4002, 4003, 5984) as defined in your `.env` file so be careful that you are not denying access to any of those components.

**Reply by gmentzik** [reply]
What I want is to block all outgoing traffic except smtp to send email and incoming traffic to VM except to port 10000 and some other from specific ip. Services like ping and ssh should be allowed of course.
Since docker overrides firewalld I am having hard time to define these rules.
I am checking the docker-user zone alternative. 
Another option I have is to create an internal docker network for all services but proxy, so only proxy has access to outside world. Would that work?
Any suggestion is welcome! 
In fact in my case it is firewalld that I need to set up.
